### PR TITLE
Setting up dev and preprod clusters in the new subscription

### PR DESCRIPTION
**Pull request by gortegaMS**
# Type of PR

- [x] Documentation changes
- [x] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [x] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

Setting up dev and preprod clusters in the new subscription

## Does this introduce a breaking change

- [x] YES ,  See https://github.com/retaildevcrews/wcnp/issues/1115,   https://github.com/retaildevcrews/wcnp/issues/1112
- [ ] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/1055 
- Closes https://github.com/retaildevcrews/wcnp/issues/1028
- References https://github.com/retaildevcrews/wcnp/issues/1115
- References https://github.com/retaildevcrews/wcnp/issues/1089

### CODE REVIEW

File: deploy/dev-ngsa-asb-westus2/config.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-    "zone": "az-westus2-dev",
-    "region": "westus2",
-    "mi.name": "ngsa-asb-ngsa-id",
-    "domain" : "cse.ms",
+    "zone": "az-westus3-dev",
+    "region": "westus",
```
Comment (AAkindele) [inline deploy/dev-ngsa-asb-westus2/config.json:3]: Should this be westus3?
```suggestion
    "region": "westus3",
```

File: monitoring/README.md [hunk 2]
```diff
@@ -194,7 +194,14 @@ You'll need to run a few more steps to completely setup the AAD Service Principa
 az keyvault set-policy --secret-permissions set --object-id $(az ad signed-in-user show --query id -o tsv) -n $ASB_KV_NAME -g $ASB_RG_CORE
 
 # Set grafana AAD client secrets
-az keyvault secret set -o table --vault-name $ASB_KV_NAME --name "grafana-aad-client-secret" --value "$CLIENT_SECRET"
+az keyvault secret set -o table --vault-name $ASB_KV_NAME --name "grafana-aad-client-secret" --value [insert CLIENT_SECRET]
```
Comment (AAkindele) [inline monitoring/README.md:197]: There are 2 `[insert CLIENT_SECRET]` sections here. Are they the same value? `CLIENT_SECRET` is set above in line 165, should that be used here?

File: scripts/clusterCreation/getClusterAdminIDforDeployment.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
```
Comment (AAkindele) [inline scripts/clusterCreation/getClusterAdminIDforDeployment.sh:1]: I don't think this file is needed anymore. Might have been from an old branch. There is similar logic in main here https://github.com/retaildevcrews/ngsa-asb/blob/main/scripts/clusterCreation/1-CheckPrerequisites.sh

File: scripts/clusterCreation/deployASB.sh [hunk 4]
```diff
@@ -0,0 +1,609 @@
+#!/bin/bash
```
Comment (AAkindele) [inline scripts/clusterCreation/deployASB.sh:1]: Should this file be here? I believe this might be from older automation work that was already merged.

**Review by AAkindele — CHANGES_REQUESTED** [review]

File: deploy/dev-ngsa-asb-westus2/config.json (hunk 1 above)
Comment (gortegaMS) [reply deploy/dev-ngsa-asb-westus2/config.json:3]: yes, thank you.

File: scripts/clusterCreation/getClusterAdminIDforDeployment.sh (hunk 3 above)
Comment (gortegaMS) [reply scripts/clusterCreation/getClusterAdminIDforDeployment.sh:1]: Actually, it referenced from file scripts/clusterCreation/2-CreateHub.sh (line 8)
But thanks for pointing that out

File: scripts/clusterCreation/deployASB.sh (hunk 4 above)
Comment (gortegaMS) [reply scripts/clusterCreation/deployASB.sh:1]: No, actually should not, thanks.

File: monitoring/README.md [hunk 5]
```diff
@@ -194,7 +197,14 @@ You'll need to run a few more steps to completely setup the AAD Service Principa
...
 
 # Set grafana AAD client secrets
-az keyvault secret set -o table --vault-name $ASB_KV_NAME --name "grafana-aad-client-secret" --value "$CLIENT_SECRET"
+az keyvault secret set -o table --vault-name $ASB_KV_NAME --name "grafana-aad-client-secret" --value $GRAFANA_OAUTH_CLIENT_SECRET
+
+az keyvault secret set -o table --vault-name $ASB_KV_NAME --name "grafana-azure-monitor-client-secret" --value $AZ_MONITOR_CLIENT_SECRET
+
+# set grafana AAD ids
+export ASB_GRAFANA_SP_CLIENT_ID=[insert CLIENT_ID]
+export ASB_GRAFANA_SP_TENANT_ID=[insert TENANT_ID]
```
Comment (AAkindele) [inline monitoring/README.md:206]: Would this be the same values set above in line 153 and 155 for `CLIENT_ID` and `TENANT_ID`?

```suggestion
export ASB_GRAFANA_SP_CLIENT_ID=$CLIENT_ID
export ASB_GRAFANA_SP_TENANT_ID=$TENANT_ID
```

File: scripts/clusterCreation/getClusterAdminIDforDeployment.sh (hunk 3 above)
Comment (gortegaMS) [reply scripts/clusterCreation/getClusterAdminIDforDeployment.sh]: Ok, Alfredo just confirmed that it is a typo and it should be referencing to scripts/clusterCreation/1-CheckPrerequisites.sh  instead,

**Review by AAkindele — APPROVED** [review]
LGTM 🚀

**Pull request merged**